crates/net/eth-wire/src/ethstream.rs

```diff
@@ -15,45 +16,45 @@ use std::{
     pin::Pin,
     task::{Context, Poll},
 };
-use tokio_stream::Stream;
+use tokio::sync::{mpsc, oneshot};
+use tokio_stream::{wrappers::UnboundedReceiverStream, Stream};
 
 /// [`MAX_MESSAGE_SIZE`] is the maximum cap on the size of a protocol message.
 // https://github.com/ethereum/go-ethereum/blob/30602163d5d8321fbc68afdcbbaf2362b2641bde/eth/protocols/eth/protocol.go#L50
 pub const MAX_MESSAGE_SIZE: usize = 10 * 1024 * 1024;
 
 /// An un-authenticated [`EthStream`]. This is consumed and returns a [`EthStream`] after the
 /// `Status` handshake is completed.
-#[pin_project]
-pub struct UnauthedEthStream<S> {
-    #[pin]
-    inner: S,
+pub struct UnauthedEthStream {
+    p2p_tx: mpsc::Sender<Bytes>,
+    p2p_rx: mpsc::UnboundedReceiver<Bytes>,
```

Comment:
I think I get the idea here.

the ethstream is wired to the p2p connection in both directions for receiving raw message bytes and delegating outgoing messages.

Reply:
precisely